### PR TITLE
Back `AliasManager` with `FCHashMap` from `StateMetadata`

### LINKED ISSUE #2968: `AliasManager` must be backed by a `FCHashMap`
### Description

It is unsafe to re-use signatures expanded from a signed state using the current `AliasManager`, because it stores the alias-to-account mapping [in a `HashMap`](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/ledger/accounts/AliasManager.java#L52).

Under heavy concurrent mutations to this `HashMap`, it is possible for [`lookupIdBy(ByteString)`](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/ledger/accounts/AliasManager.java#L161) to return `MISSING_NUM` in `expandSignatures()` **even for an account whose alias has not actually been removed**. This means the [normal precautions taken in `Rationalization.performFor(TxnAccessor)`](https://github.com/hashgraph/hedera-services/blob/master/hedera-node/src/main/java/com/hedera/services/sigs/Rationalization.java#L85) will be of no use, because the `SigImpactHistorian` will (correctly) report that no linked aliases have changed.

So any node that got "unlucky" during `expandSignatures()` will resolve to `INVALID_ACCOUNT_ID` a transfer that would have otherwise resolved to `SUCCESS`. More generally, when answering queries from a signed state, we need access to exactly the aliases that existed for that state.

### Steps to reproduce

Not trivial to reproduce, need to run clients that create and use aliased accounts in large volume and material TPS, but you will eventually get an ISS.

### Additional context

_No response_

### Hedera network

other

### Version

v0.24.0-alpha.3

### Operating system

_No response_

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Related issue(s)**:
- Closes #2968 

**Description**:
 - Adds a [`FCHashMap<ByteString, EntityNum> aliases` to the `StateMetadata`](https://github.com/hashgraph/hedera-services/blob/02968-M-BackAliasMgrWithFCHashMap/hedera-node/src/main/java/com/hedera/services/state/org/StateMetadata.java#L34) to capture a view of aliases as they exist in each fast-copy of the world state.
 - Updates `MutableStateChildren` [to maintain a reference](https://github.com/hashgraph/hedera-services/blob/02968-M-BackAliasMgrWithFCHashMap/hedera-node/src/main/java/com/hedera/services/context/MutableStateChildren.java#L68) to such an `aliases` map.
     * Removes the unnecessary `StateAccessor` class that used to wrap `MutableStateChildren`.
 - Updates the [`AliasManager` singleton used by the working state](https://github.com/hashgraph/hedera-services/blob/02968-M-BackAliasMgrWithFCHashMap/hedera-node/src/main/java/com/hedera/services/ledger/accounts/AliasManager.java#L57) to continuously use the [latest copy of the `aliases` map](https://github.com/hashgraph/hedera-services/blob/02968-M-BackAliasMgrWithFCHashMap/hedera-node/src/main/java/com/hedera/services/state/StateModule.java#L327).
 - Switches the [`StateChildrenSigMetadataLookup`](https://github.com/hashgraph/hedera-services/blob/02968-M-BackAliasMgrWithFCHashMap/hedera-node/src/main/java/com/hedera/services/sigs/metadata/StateChildrenSigMetadataLookup.java#L78) to construct its own `AliasManager` from its given `StateChildren`.
     * 🎉  Now when the `SigReqsManager` is [able to expand signatures from a signed, initialized state](https://github.com/hashgraph/hedera-services/blob/02968-M-BackAliasMgrWithFCHashMap/hedera-node/src/main/java/com/hedera/services/sigs/order/SigReqsManager.java#L137), any required aliases lookups will be free from thread interference (concurrent mutations in `handleTransaction()` will only target the mutable copy of the aliases `FCHashMap`). 

Also does a bit of house-keeping:
 - Deletes a pointless legacy `AtomicCounter` class.
 - Eliminates redundant use of the `TokenStore` in `StateView`---all the necessary information is available from `stateChildren.tokens()`.
     * This diff is the tokens analogue of what should be done for schedules in https://github.com/hashgraph/hedera-services/issues/2972.

### CODE REVIEW

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Review by Neeharika-Sompalli — COMMENTED** [review]
Merge conflict to be resolved

File: hedera-node/src/main/java/com/hedera/services/state/org/StateMetadata.java [hunk 1]
```diff
@@ -51,10 +57,14 @@ public StateMetadata copy() {
...
 	public void release() {
-		// No-op
+		aliases.release();
 	}
 
 	public ServicesApp app() {
 		return app;
 	}
+
+	public FCHashMap<ByteString, EntityNum> aliases() {
```
Comment (cody-l) [inline hedera-node/src/main/java/com/hedera/services/state/org/StateMetadata.java:67]: Nit, better for return type to be `Map<ByteString, EntityNum>` unless caller really needs to do some operations that are only possible on an `FCHashMap` type. Same goes for sibling methods in `ServicesState.java` and `StateMetadata.java`, and the constructor in `AliasManager.java`.

File: hedera-node/src/test/java/com/hedera/services/fees/calculation/consensus/queries/GetMerkleTopicInfoResourceUsageTest.java [hunk 2]
```diff
@@ -64,7 +64,7 @@ void setup() {
 		topics = mock(MerkleMap.class);
 		final var children = new MutableStateChildren();
 		children.setTopics(topics);
-		view = new StateView(null, null, children, null);
+		view = new StateView( null, children, null);
```
Comment (cody-l) [inline hedera-node/src/test/java/com/hedera/services/fees/calculation/consensus/queries/GetMerkleTopicInfoResourceUsageTest.java:67]: Nit, remove space from before `null`

File: hedera-node/src/main/java/com/hedera/services/context/MutableStateChildren.java [hunk 3]
```diff
@@ -207,16 +193,22 @@ public RecordsRunningHashLeaf runningHashLeaf() {
 		return refRunningHashLeaf;
 	}
 
-	public void setRunningHashLeaf(RecordsRunningHashLeaf runningHashLeaf) {
-		this.runningHashLeaf = new WeakReference<>(runningHashLeaf);
+	@Override
+	public FCHashMap<ByteString, EntityNum> aliases() {
+		final var refAliases = aliases.get();
+		Objects.requireNonNull(refAliases);
```
Comment (cody-l) [inline hedera-node/src/main/java/com/hedera/services/context/MutableStateChildren.java:199]: A nifty trick I like using, `Objects.requireNonNull` actually returns the value, so you can make this a one liner.

`final var refAliases = Objects.requireNonNull(refAliases);`

**Review by cody-l — APPROVED** [review]

File: hedera-node/src/main/java/com/hedera/services/state/org/StateMetadata.java (hunk 1 above)
Comment (tinker-michaelj) [reply hedera-node/src/main/java/com/hedera/services/state/org/StateMetadata.java]: Ah excellent point, only `StateMetadata` actually needs to know that it's dealing with an `FCHashMap`. Fixed.

File: hedera-node/src/test/java/com/hedera/services/fees/calculation/consensus/queries/GetMerkleTopicInfoResourceUsageTest.java (hunk 2 above)
Comment (tinker-michaelj) [reply hedera-node/src/test/java/com/hedera/services/fees/calculation/consensus/queries/GetMerkleTopicInfoResourceUsageTest.java:67]: Fixed.

**Pull request merged**